### ISSUE TITLE
Model dependency on how input files are split in preprocessing

### ISSUE THREAD

**Issue by alpk** [Pointcept/Pointcept#177]
Hello @Gofinge, we are using ptv3 for semantic segmentation with our custom city segmentation pointcloud dataset on buildings, plants, road and water classes similar to SensatUrban.

Our each file contains only one object hence only one label. We divide our each pcd file into fixed point size (such as 20k) smaller chunks. For example 43k points divided into 20k-20k-3k points before feeding to the model. (Due to our small gpu size)
With this approach we have satisfactory results on testing on cities not included in the training set, these test files are also divided based on the aforementioned strategy. 

We have realized even though we have a good model, when we try to run inference on non-annotated dataset (we gave garbage labels then convert to pth file) our results are also garbage. When we change the fixed point sizes we are getting different results for each run.

Have you noticed anything about dependency on input size? What could be the best strategy here, should we put multiple labels into one file then feed to the model?

**Comment by Gofinge** [comment]
That's a good question. Here are my suggestions, and if any misunderstanding exists, please tell me.
> What could be the best strategy here, should we put multiple labels into one file and then feed it to the model?

You mentioned that each point cloud sample for training and standard evaluation only contains one label, which means the task is not semantic segmentation. It is object classification (refer the ModelNet40 and ScanObjNN, one category for one sample).

You said _"when we try to run inference on any non-annotated dataset (we gave garbage labels then convert to pth file) our results are also garbage"_. Do you mean multiple labels in one point cloud?  If so, the task is already changed from classification to segmentation which might cause a "garbage" performance.

> For example 43k points divided into 20k-20k-3k points before feeding to the model.
> Have you noticed anything about dependency on input size?

If you split point clouds into multiple fixed chunks, some small split point clouds will not contain the necessary information to judge the category. So I don't recommend this strategy. There are many ways to solve the issue. 1. Increase grid size for grid sampling. Typically perception tasks do not need so many points. 2. Use random SphereCrop during training, which you can make sure each time you feed your network with sub-point clouds containing points numbered under maybe 20k (It is a really small number, of PTv3, can process a much larger number of points without a huge increase of memory cost).

**Comment by alpk** [comment]
Let me clear what we have during training and testing, 

We have a big pointcloud file and our annotators cut each object into seperate files. Each file is actually one instance at this point for the sake of simplicity. Normally we concatenate them for visualization to check our labels.

For training we didn't bother to concat these files but divided them into seperate chunks as mentioned above. So during training, model gets different chunks of different objects (we can get at least 12 for training batch size)

During validation and precise testing we again use the same file structure. For example, we cut part of the city and provide it during testing and we get good results. But what we test is actually one label per file again, so model achieves very high accuracy.

As you mentioned we might have converted the segmentation problem into a classification problem. Each of our files have  clear boundary cuts, since object boundaries must also be covered by the model and not the input files.
Getting garbage outputs probably is the result of not providing the boundaries during training.

We are currently trying to divide the area into grids. We concatenate the annotated files of a city and apply grid cut on them and make sure each grid part contains multiple labels and boundaries. Since the datasets are not huge hgowever, we are not sure if this operation will introduce other dependencies to grid shape inputs and their parameters such as grid size. 
We will work on your suggestions and let you know.

Thank you for the helpful insights.
